.github/workflows/test_on_push.yml

```diff
@@ -131,6 +139,18 @@ jobs:
         - name: Install nox
           uses: wntrblm/nox@2024.04.15
 
-        - name: Test the generated project
-          working-directory: ./pybamm-example-project
+        - name: Test the generated project with default hatch build backend and no vcs
+          working-directory: ../pybamm-example-project
+          run: nox -s generated-project-tests
+
+        - name: Test the generated project with setuptools build backend and no scm
+          working-directory: ../pybamm-with-setuptools
+          run: nox -s generated-project-tests
+
+        - name: Test the generated project with setuptools build backend and scm
+          working-directory: ../pybamm-with-setuptools-scm
+          run: nox -s generated-project-tests
+
+        - name: Test the generated project with hatch build backend and vcs
+          working-directory: ../pybamm-with-hatch-vcs
```

Comment:
I'd really like to make these separate jobs so that they don't depend on the previous step and can run parallelly (making everything cleaner).

Reply:
Yes, a matrix configuration would be good. You can use the following:

```yaml
matrix:
  build-backend: [hatchling, setuptools]
  versioning: [with-vcs, without-vcs]
```

and include `{{ matrix.build-backend }}-{{ matrix.versioning }}` in the job name. Feel free to play around with the variable names, of course!

Reply:
I added them for better coverage, which in total would be 16x4 combinations of runners 16 of which are the combination of discrete Operating systems and python versions and the 4 combinations are of the build backends and there respective vcs.